bindings/src/main/java/io/pravega/storage/filesystem/FileSystemStorage.java

```diff
@@ -319,6 +319,7 @@ private Void doWrite(SegmentHandle handle, long offset, InputStream data, int le
                     totalBytesWritten += bytesWritten;
                     length -= bytesWritten;
                 }
+                channel.force(true);
```

Comment:
Can you justify why we also need to force a metadata sync? You need to reason about what part of the metadata you are actually updating.

Reply:
I think you are right. We don't need to explicitly synchronize metadata to update file size on disk, it should be already done as part of data sync itself.

Reply:
doublechecked fdatasync() documentation, filesize changes automatically trigger metadata flush . 

> It does not flush modified metadata _unless that metadata is needed in order to allow a subsequent data retrieval to be correctly handled_. On the other hand, a change to the file size, would require a metadata flush.